File: code/algorithms/sorting/basicsort/bubble.go
package basicsort

type BubbleSort struct {
	BasicSort
}

func (s *BubbleSort) Sort(a []int) {
	for i := 0; i < len(a); i++ {
		for j := 0; j < len(a)-1; j++ {
			if a[j] > a[j+1] {
				s.swap(a, j, j+1)
			} else {
				s.pass(j, j+1)
			}
		}
	}
}

File: notes/.vuepress/components/algoman/algorithms/sorting/bubble.js
import Sorter from './sorter'
export default class BubbleSorter extends Sorter {
    constructor() {
        super()
    }

    sort(array) {
        this.array = array
        this.commands = []
        for (var i = 0; i < this.array.length; i++) {
            for (var j = i + 1; j < this.array.length; j++) {
                if (this.array[i] > this.array[j]) {
                    this.swap(i, j)
                }
            }
        }
    }
}
